### PR TITLE
Add matplotlib and WCSAxes to list of optional dependencies in install docs (fixes #3608)

### LINKED ISSUE #3608: Add matplotlib and WCSAxes to list of optional dependencies in install docs
It is used in astropy.visualization.

Also add WCSAxes since it's used when WCS is used as a projection in Matplotlib.

### PR DESCRIPTION
**Pull request by mtzhisham**
added matloptlib and WCSAxes to optional dependencies list 
https://github.com/astropy/astropy/issues/3608

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -29,6 +29,11 @@ Astropy also depends on other packages for optional features:
 
 - `xmllint <http://www.xmlsoft.org/>`_: To validate VOTABLE XML files.
 
+- `matplotlib <http://matplotlib.org/>`_: To plot 2D data.
```
Comment (eteq) [inline docs/install.rst]: Maybe change this from ``to plot 2D data`` to just something like ``To provide plotting functionality that `astropy.visualization` echances ``? It's not exactly 2D *data* that's being plotted, but rather *2D plots*.

**Comment by eteq** [comment]
One other tip @mtzhisham : if you put the text ``closes #3608`` (or ``fixes #3608``, or similar) to the description of your PR, it will automatically close #3608 when this gets merged.  Not critical to do this now because we can always close it manually, but it makes things slightly easier to manage.

**Comment by mtzhisham** [comment]
@eteq read the documentation for both and updated

**Comment by astrofrog** [comment]
Thanks @mtzhisham!

**Comment by embray** [comment]
Even for documentation updates please provide a milestone before merging, so we can make sure the doc update is included in the next release, where appropriate.

**Pull request merged**